modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticDruckerPrager.C

```diff
@@ -0,0 +1,275 @@
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "TensorMechanicsPlasticDruckerPrager.h"
+
+template<>
+InputParameters validParams<TensorMechanicsPlasticDruckerPrager>()
+{
+  InputParameters params = validParams<TensorMechanicsPlasticModel>();
+  MooseEnum mc_interpolation_scheme("outer_tip=0 inner_tip=1 lode_zero=2 inner_edge=3 native=4", "lode_zero");
+  params.addParam<MooseEnum>("mc_interpolation_scheme", mc_interpolation_scheme, "Scheme by which the Drucker-Prager cohesion, friction angle and dilation angle are set from the Mohr-Coulomb parameters mc_cohesion, mc_friction_angle and mc_dilation_angle.  Consider the DP and MC yield surfaces on the devatoric (octahedral) plane.  Outer_tip: the DP circle touches the outer tips of the MC hex.  Inner_tip: the DP circle touches the inner tips of the MC hex.  Lode_zero: the DP circle intersects the MC hex at lode angle=0.  Inner_edge: the DP circle is the largest circle that wholey fits inside the MC hex.  Native: The DP cohesion, friction angle and dilation angle are set equal to the mc_ parameters entered.");
+  params.addRequiredParam<UserObjectName>("mc_cohesion", "A TensorMechanicsHardening UserObject that defines hardening of the Mohr-Coulomb cohesion.  Physically this should not be negative.");
+  params.addRequiredParam<UserObjectName>("mc_friction_angle", "A TensorMechanicsHardening UserObject that defines hardening of the Mohr-Coulomb friction angle (in radians).  Physically this should be between 0 and Pi/2.");
+  params.addRequiredParam<UserObjectName>("mc_dilation_angle", "A TensorMechanicsHardening UserObject that defines hardening of the Mohr-Coulomb dilation angle (in radians).  Usually the dilation angle is not greater than the friction angle, and it is between 0 and Pi/2.");
+  params.addClassDescription("Non-associative Drucker Prager plasticity with no smoothing of the cone tip.");
+  return params;
+}
+
+TensorMechanicsPlasticDruckerPrager::TensorMechanicsPlasticDruckerPrager(const InputParameters & parameters) :
+    TensorMechanicsPlasticModel(parameters),
+    _mc_cohesion(getUserObject<TensorMechanicsHardeningModel>("mc_cohesion")),
+    _mc_phi(getUserObject<TensorMechanicsHardeningModel>("mc_friction_angle")),
+    _mc_psi(getUserObject<TensorMechanicsHardeningModel>("mc_dilation_angle")),
+    _mc_interpolation_scheme(getParam<MooseEnum>("mc_interpolation_scheme")),
+    _zero_cohesion_hardening(_mc_cohesion.modelName().compare("Constant") == 0),
+    _zero_phi_hardening(_mc_phi.modelName().compare("Constant") == 0),
+    _zero_psi_hardening(_mc_psi.modelName().compare("Constant") == 0)
+{
+  if (_mc_phi.value(0) < 0 || _mc_psi.value(0) < 0)
+    mooseError("TensorMechanicsPlasticDruckerPrager: MC friction and dilation angles must lie in [0, Pi/2]");
```

Comment:
The check and the error message are saying something different (you are not checking the pi/2 part)